src/gapp_login/wechat/endpoints.py

```diff
@@ -0,0 +1,86 @@
+import logging
+from urllib import parse
+
+from authlib_gino.fastapi_session.api import auth, login_context
+from authlib_gino.fastapi_session.models import User
+from fastapi import APIRouter, HTTPException, Form, FastAPI, Depends
+from starlette import status
+from starlette.requests import Request
+
+from . import config
+from .models import db, WeChatIdentity
+from .utils import WeChatAuthError
+
+log = logging.getLogger(__name__)
+router = APIRouter()
+
+
+@router.put("/login/wechat")
+async def login_wechat(
+    request: Request, code: str = Form(...), ctx=Depends(login_context)
+):
+    idp = ctx["idp"]
+    idp_params = dict(parse.parse_qsl(parse.unquote(ctx["idp_params"])))
```

Comment:
I feel like unquote is unnecessary - it was already unquoted before creating the JWT.

Reply:
fixed